pgcli/pgcompleter.py

```diff
@@ -100,6 +102,15 @@ def __init__(self, smart_completion=True, pgspecial=None, settings=None):
         self.call_arg_oneliner_max = settings.get("call_arg_oneliner_max", 2)
         self.search_path_filter = settings.get("search_path_filter")
         self.generate_aliases = settings.get("generate_aliases")
+
+        # when should this file be loaded? IO in constructors is not my preference but slow startup is
+        # probably better than slow first query
+        alias_map_file = settings.get("alias_map_file")
+        if alias_map_file is not None:
+            with open(alias_map_file) as fo:
+                self.alias_map = json.load(fo)
```

Comment:
I wanted to flag this part as I wasn't sure of what approaches here fir the project goals

1. If the file doesn't exist what should I do? Allow the crash or just handle it silently and set the map to `None`?
2. is loading the file in the constructor a good idea? I tend not to do IO in constructors normally but since theoretically the alias file could be quite large I thought slow startup was better than a slow first query. Maintainers might think the opposite

Reply:
1. If someone specified a file in the config, and it doesn't exist, it should be an error. We can assume that the user wanted the file to be picked up, and so the user will be interested in fixing the path.
2. How large is "quite large"? Did you do any benchmarks with your file? If you think it's worth the effort, you could run a background thread to load these aliases, like the [completion refresher](https://github.com/dbcli/pgcli/blob/main/pgcli/completion_refresher.py) does.

Reply:
I pushed a new version that wraps loading/decoding into a helper function and raises an informative error if the map_file cannot be read


My personal alias file is tiny - 5 aliases and the whole file is ~220 bytes - and so for myself at least loading is fast

```
In [1]: from pgcli.pgcompleter import load_alias_map_file

In [2]: from pgcli import config

In [3]: import timeit

In [4]: map_file = config.get_config()["main"]["alias_map_file"]

In [5]: t = timeit.Timer(lambda: load_alias_map_file(map_file))

In [6]: t.timeit(10_000)
Out[6]: 0.15082729199999534
```

I tried upping the alias file to 1000 aliases and it's still  pretty fast

```
In [9]: t = timeit.Timer(lambda: load_alias_map_file(map_file))

In [10]: t.timeit(10_000)
Out[10]: 1.3473110840000118
```

> How large is "quite large"? 

For my own usage I can't see the file getting bigger than 200 lines (we have ~ 120 tables in prod) which is 8kb and quick to load

```
In [8]: t.timeit(10_000)
Out[8]: 0.3809067500000083
```

my caution was purely down to experience of users always doing something surprising and using features in unexpected ways.

I don't think that loading the aliases would need to be performed in a background thread